### PR TITLE
Dynamic map for the GUI

### PR DESCRIPTION
**Pull request by VivaanBahl**
Is able to work both online and offline, has a much easier interface for adding data points, can zoom and (kinda) pan around

### CODE REVIEW

**Comment by TrevorDecker** [comment]
DO NOT MERGE


Please do not commit log files into git repo

**Comment by ramensandwich** [comment]
Are log files not part of .gitignore?

**Comment by VivaanBahl** [comment]
Whoops didn't realize I'd committed those

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java [hunk 1]
```diff
@@ -2,125 +2,181 @@
 
 import java.awt.Color;
 import java.awt.Graphics;
-import java.awt.Graphics2D;
+import java.awt.event.KeyEvent;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java]: Codacy Issue found: [Avoid unused imports such as 'java.awt.event.KeyEvent'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1397684331/issues/source?bid=2938665&fileBranchId=2982055#l5)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java [hunk 2]
```diff
@@ -2,125 +2,181 @@
...
+import java.awt.event.KeyListener;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java]: Codacy Issue found: [Avoid unused imports such as 'java.awt.event.KeyListener'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1397684331/issues/source?bid=2938665&fileBranchId=2982055#l6)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java (hunk 1 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java]: Codacy Issue found: [Avoid unused imports such as 'java.awt.event.KeyEvent'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1408050394/issues/source?bid=2938665&fileBranchId=2982055#l5)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java (hunk 2 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java]: Codacy Issue found: [Avoid unused imports such as 'java.awt.event.KeyListener'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1408050394/issues/source?bid=2938665&fileBranchId=2982055#l6)

**Comment by TrevorDecker** [comment]
I was able to break it, but for the first 30 seconds it seemed awesome. 

I was zooming in and out when it crashed

system output:

Stable Library
=========================================
Native lib Version = RXTX-2.2pre2
Java lib Version   = RXTX-2.1-7
WARNING:  RXTX Version mismatch
	Jar version = RXTX-2.1-7
	native lib Version = RXTX-2.2pre2
[/dev/tty.Bluetooth-Incoming-Port, /dev/cu.Bluetooth-Incoming-Port]
Starting GUI
H
Populated!
Starting Robot
Starting Logging
Logic Exception Setup properly
Initializing GPS Serial Connection
Initializing IMU Serial Connection
Initializing Encoder Serial Connection
rbsmnode start
Initializing Vision System
JUST KIDDING VISION SYSTEM CANNOT WORK NOW

Bing: attribution data is not yet loaded.

Starting Logging
initializing the SensorPlayer
File doesn't exist!
Starting to read from file!
Error while trying to read from the playback file!
Bing: attribution data is not yet loaded.
Bing: attribution data is not yet loaded.
Bing: attribution data is not yet loaded.
Bing: attribution data is not yet loaded.
Bing: attribution data is not yet loaded.
Successfully loaded Bing attribution data.
Failed loading http://ecn.t0.tiles.virtualearth.net/tiles/a.jpeg?g=4739: class j
ava.io.IOException: Server returned HTTP response code: 400 for URL: http://ecn.
t0.tiles.virtualearth.net/tiles/a.jpeg?g=4739
shutting down Robot

**Comment by VivaanBahl** [comment]
darn it :(

**Comment by TrevorDecker** [comment]
Also can you catch all of the warnings and errors that it produces to logicErrors so that they do not pollute the output stream.

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java (hunk 1 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java]: Codacy Issue found: [Avoid unused imports such as 'java.awt.event.KeyEvent'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1558838655/issues/source?bid=2938665&fileBranchId=2982055#l5)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java (hunk 2 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java]: Codacy Issue found: [Avoid unused imports such as 'java.awt.event.KeyListener'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1558838655/issues/source?bid=2938665&fileBranchId=2982055#l6)

**Comment by TrevorDecker** [comment]
Any luck fixing the error posted above?

**Comment by TrevorDecker** [comment]
bump

**Comment by TrevorDecker** [comment]
this feature  has been merged into trevor test,  we will get this feature into master when trevor test is merged in

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java (hunk 1 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java]: Codacy Issue found: [Avoid unused imports such as 'java.awt.event.KeyEvent'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1558838655/issues/source?bid=2938665&fileBranchId=2982055#l5)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java (hunk 2 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java]: Codacy Issue found: [Avoid unused imports such as 'java.awt.event.KeyListener'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1558838655/issues/source?bid=2938665&fileBranchId=2982055#l6)

**Pull request closed without merging**